scriptmodules/ports/wolf4sdl.sh

```diff
@@ -30,11 +30,31 @@ function get_opts_wolf4sdl() {
     echo 'wolf4sdl-spear -DCARMACIZED -DGOODTIMES -DSPEAR' # spear of destiny
 }
 
-function get_bins_wolf4sdl() {
-    local opt
-    while read -r opt; do
-        echo "${opt%% *}"
-    done < <(get_opts_wolf4sdl)
+function _add_games_wolf4sdl() {
+    local cmd="$1"
+    declare -A games=(
+        ['vswap.wl1']="Wolfenstein 3D demo"
+        ['vswap.wl6']="Wolfenstein 3D"
+        ['vswap.sd1']="Wolfenstein 3D - Spear of Destiny Ep 1"
+        ['vswap.sd2']="Wolfenstein 3D - Spear of Destiny Ep 2"
+        ['vswap.sd3']="Wolfenstein 3D - Spear of Destiny Ep 3"
+    )
+    local game
+    local wad
+
+    # remove obsolete emulator entries
+    if [[ "$md_mode" == "remove" ]]; then
+        for game in $(get_opts_wolf4sdl; echo -e "wolf4sdl-spear2\nwolf4sdl-spear3"); do
```

Comment:
this is wrong - as it will loop through `wolf4sdl-sw-v14 -DCARMACIZED -DUPLOAD` etc

Reply:
It's what I intended. The next line filters ```$game``` to omit anything after space, like the old ```get_bins_wolf4sdl```. I just did it that way to avoid assigning a new var once.

Reply:
Ah do you mean it loops through each phrase on a newline? I'll check.

Reply:
Yep. The other loop didn't but that one will.

Reply:
It will call addPort with 

```
wolf4sdl-sw-v14
-DCARMACIZED
-DUPLOAD
wolf4sdl-3dr-v14
-DCARMACIZED
wolf4sdl-gt-v14
-DCARMACIZED
-DGOODTIMES
wolf4sdl-spear
-DCARMACIZED
-DGOODTIMES
-DSPEAR
wolf4sdl-spear2
wolf4sdl-spear3
```

Reply:
Yes, you're right. If you can hang on a bit, I'll do some research to see if we can do anything to circumvent the problem with the return interfering with process substitution loop.